Tasks/KubernetesManifestV0/src/actions/deploy.ts

```diff
@@ -0,0 +1,193 @@
+                    promises.push(checkRolloutStatus(words[0], words[1].trim()));
+                }
+            });
+            types.forEach(line => {
+                let words = line.split(" ");
+                if (recognizedType.filter(type => words[0].startsWith(type)).length > 0) {
+                    promises.push(annotateResourceSets(words[0], words[1].trim()));
+                }
+            });
+            Q.all(promises).then(() => {
+                console.log("all done")
+            })
+        }
+    });
+};
+
+function getCommandConfigurationFiles(filePattern: string, containers): string[] {
+    if (filePattern.length == 0) {
+        return [""];
+    }
+    var files: string[] = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory") || process.cwd(), filePattern);
+    if (!files || !files.length) {
+        return [];
+    }
+
+    if (containers != []) {
+        let newFilePaths = [];
+        const tempDirectory = utils.getTempDirectory();
+        files.forEach((filePath: string) => {
+            var contents = fs.readFileSync(filePath).toString();
+            containers.forEach((container: string) => {
+                let imageName = container.split(":")[0];
+                if (contents.indexOf(imageName) > 0) {
+                    contents = replaceAll(contents, imageName, container);
+                }
+            });
+
+            let fileName = path.join(tempDirectory, path.basename(filePath));
+            fs.writeFileSync(
+                path.join(fileName),
+                contents
+            );
+
+            newFilePaths.push(fileName);
+        });
+    }
+
+    return files;
+}
+
+function replaceAll(currentString: string, replaceToken, replaceValue) {
+    let i = currentString.indexOf(replaceToken);
+    if (i < 0) {
+        return currentString;
+    }
+
+    let newString = currentString.substring(0, i);
+    let leftOverString = currentString.substring(i);
+    newString += replaceValue + leftOverString.substring(leftOverString.indexOf("\n"));
+    if (newString == currentString) {
+        return newString;
+    }
+    return replaceAll(newString, replaceToken, replaceValue);
+}
+
+function apply(configurationPath: string) {
+    var command = tl.tool(tl.which("kubectl"));
+    command.arg("apply");
+    command.arg(getNameSpace());
+    command.arg(["-f", configurationPath]);
+    return command.execSync();
+}
+
+function annotate(configurationPath: string): any {
+    var command = tl.tool(tl.which("kubectl"));
+    command.arg("annotate");
+    command.arg(getNameSpace());
+    command.arg(["-f", configurationPath]);
+    command.arg(`azure-pipelines/execution=${tl.getVariable("Build.BuildNumber")}`)
+    command.arg(`azure-pipelines/pipeline="${tl.getVariable("Build.DefinitionName")}"`)
+    command.arg(`azure-pipelines/executionuri=${tl.getVariable("System.TeamFoundationCollectionUri")}_build/results?buildId=${tl.getVariable("Build.BuildId")}`)
```

Comment:
duplicate code in annotate and annotateResourceSets,.,,

Reply:
changed